tests/helpers/mock-process.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+let td = require('testdouble');
+let EventEmitter = require('events');
+
+module.exports = class FakeProcess extends EventEmitter {
+  constructor(options) {
+    super();
+
+    options = options || {};
+
+    const stdin = Object.assign(new EventEmitter(), {
+      isRaw: false,
+      setRawMode: td.function(),
+    }, options.stdin || {});
+
+    const topLevelProps = Object.assign({
+      platform: 'MockOS',
+      exit: td.function(),
+    }, options);
+
+    Object.assign(this, topLevelProps, { stdin });
```

Comment:
This is in incomplete form right now. For example there is no `stdout`. Currently we have only what we actually use.